crates/pico_core/src/key.rs

```diff
@@ -0,0 +1,18 @@
+use std::hash::{DefaultHasher, Hash, Hasher};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Key(u64);
```

Comment:
can we rename this something more intuitive? I don't yet understand what it is doing. It's an id of a memoized function, ...right?

Reply:
sgtm. Let's add that as a comment!